File: tests/fixtures/java_project/Calculator.java
package com.example.calculator;

import java.util.List;
import java.util.ArrayList;
import java.util.function.BiFunction;

/**
 * Calculator class with basic arithmetic operations
 */
public class Calculator {
    
    /**
     * Adds two integers
     * @param a first number
     * @param b second number
     * @return the sum
     */
    public static int add(int a, int b) {
        return a + b;
    }
    
    /**
     * Subtracts b from a
     * @param a the minuend
     * @param b the subtrahend
     * @return the difference
     */
    public static int subtract(int a, int b) {
        return a - b;
    }
    
    /**
     * Multiplies two numbers
     */
    public static int multiply(int a, int b) {
        return a * b;
    }
    
    /**
     * Divides a by b
     * @throws ArithmeticException if b is zero
     */
    public static double divide(int a, int b) {
        if (b == 0) {
            throw new ArithmeticException("Division by zero");
        }
        return (double) a / b;
    }
    
    /**
     * Applies a custom operation to two numbers
     */
    public static int applyOperation(int a, int b, BiFunction<Integer, Integer, Integer> operation) {
        int result = operation.apply(a, b);
        System.out.println("Result is: " + result);
        return result;
    }
    
    private static int privateHelper(int x) {
        return x >= 0 ? x : -x;
    }
}

/**
 * A calculator class that maintains state
 */
class CalculatorState {
    private int value;
    private List<String> history;
    
    /**
     * Creates a new calculator with value 0
     */
    public CalculatorState() {
        this.value = 0;
        this.history = new ArrayList<>();
    }
    
    /**
     * Creates a calculator with an initial value
     */
    public CalculatorState(int initialValue) {
        this.value = initialValue;
        this.history = new ArrayList<>();
    }
    
    /**
     * Gets the current value
     */
    public int getValue() {
        return value;
    }
    
    /**
     * Sets the current value
     */
    public void setValue(int value) {
        this.value = value;
    }
    
    /**
     * Adds a number to the current value
     */
    public int add(int n) {
        value += n;
        history.add("add " + n);
        return value;
    }
    
    /**
     * Subtracts a number from the current value
     */
    public int subtract(int n) {
        value -= n;
        history.add("subtract " + n);
        return value;
    }
    
    /**
     * Resets the calculator to zero
     */
    public void reset() {
        value = 0;
        history.clear();
    }
    
    /**
     * Gets the operation history
     */
    public List<String> getHistory() {
        return new ArrayList<>(history);
    }
    
    /**
     * Checks if calculator has any history
     */
    public boolean hasHistory() {
        return !history.isEmpty();
    }
    
    @Override
    public String toString() {
        return "Calculator(value: " + value + ")";
    }
    
    private void logOperation(String operation) {
        history.add(operation);
    }
}

File: tests/fixtures/csharp_project/Calculator.cs
/// <summary>
/// Calculator module with basic arithmetic operations
/// </summary>
namespace CalculatorApp
{
    using System;
    using CalculatorApp.Models;

    /// <summary>
    /// Provides basic arithmetic operations
    /// </summary>
    public static class Calculator
    {
        /// <summary>
        /// Adds two numbers together
        /// </summary>
        /// <param name="a">First number</param>
        /// <param name="b">Second number</param>
        /// <returns>The sum of a and b</returns>
        public static int Add(int a, int b)
        {
            return a + b;
        }

        /// <summary>
        /// Subtracts b from a
        /// </summary>
        /// <param name="a">The minuend</param>
        /// <param name="b">The subtrahend</param>
        /// <returns>The difference a - b</returns>
        public static int Subtract(int a, int b)
        {
            return a - b;
        }

        /// <summary>
        /// Multiplies two numbers
        /// </summary>
        public static int Multiply(int a, int b)
        {
            return a * b;
        }

        /// <summary>
        /// Divides a by b, returns null if b is zero
        /// </summary>
        public static int? Divide(int a, int b)
        {
            if (b == 0)
                return null;
            return a / b;
        }

        /// <summary>
        /// Applies a custom operation to two numbers
        /// </summary>
        public static int ApplyOperation(int a, int b, Func<int, int, int> operation)
        {
            int result = operation(a, b);
            Func<int, string> formatter = x => $"Result is: {x}";
            Console.WriteLine(formatter(result));
            return result;
        }

        /// <summary>
        /// Creates a new calculator instance
        /// </summary>
        public static CalculatorState CreateCalculator()
        {
            return new CalculatorState();
        }

        /// <summary>
        /// Creates a calculator with an initial value
        /// </summary>
        public static CalculatorState CreateCalculatorWithValue(int value)
        {
            return new CalculatorState(value);
        }

        /// <summary>
        /// Calculates the distance between two points
        /// </summary>
        public static double PointDistance(Point p1, Point p2)
        {
            double dx = p1.X - p2.X;
            double dy = p1.Y - p2.Y;
            return Math.Sqrt(dx * dx + dy * dy);
        }

        private static int PrivateHelper(int x)
        {
            return x >= 0 ? x : -x;
        }
    }

    /// <summary>
    /// A simple calculator class that maintains state
    /// </summary>
    public class CalculatorState
    {
        private List<string> _history;

        /// <summary>
        /// Gets or sets the current value
        /// </summary>
        public int Value { get; set; }

        /// <summary>
        /// Gets whether the calculator has any history
        /// </summary>
        public bool HasHistory => _history.Count > 0;

        /// <summary>
        /// Creates a new calculator with value 0
        /// </summary>
        public CalculatorState()
        {
            Value = 0;
            _history = new List<string>();
        }

        /// <summary>
        /// Creates a new calculator with the specified initial value
        /// </summary>
        public CalculatorState(int initialValue)
        {
            Value = initialValue;
            _history = new List<string>();
        }

        /// <summary>
        /// Adds a number to the current value
        /// </summary>
        public int Add(int n)
        {
            Value += n;
            _history.Add($"add {n}");
            return Value;
        }

        /// <summary>
        /// Subtracts a number from the current value
        /// </summary>
        public int Subtract(int n)
        {
            Value -= n;
            _history.Add($"subtract {n}");
            return Value;
        }

        /// <summary>
        /// Resets the calculator to zero
        /// </summary>
        public void Reset()
        {
            Value = 0;
            _history.Clear();
        }

        /// <summary>
        /// Gets the operation history
        /// </summary>
        public List<string> GetHistory()
        {
            return new List<string>(_history);
        }

        private void LogOperation(string operation)
        {
            _history.Add(operation);
        }

        public override string ToString()
        {
            return $"Calculator(value: {Value})";
        }
    }
}
